test/test_sparse.py

```diff
@@ -630,6 +636,9 @@ def test_tensor(x):
         x = torch.sparse.FloatTensor(2, 3, 4)
         test_tensor(x)
 
+        x = torch.sparse.HalfTensor(2, 3, 4)
```

Comment:
looks like we may need a test for the `torch.sparse.cuda.HalfTensor` constructor?

Reply:
added 
```
        x = torch.cuda.sparse.HalfTensor(2, 3, 4)
        test_tensor(x)
```